android/app/src/main/java/com/on/turip/ui/search/regionresult/RegionResultActivity.kt

```diff
@@ -92,6 +96,24 @@ class RegionResultActivity : BaseActivity<ActivityRegionResultBinding>() {
         viewModel.serverError.observe(this) {
             updateUIVisibility()
         }
+
+        lifecycleScope.launch {
+            viewModel.uiEvent.collect { event ->
+                when (event) {
+                    CommonEvent.TokenExpiration -> navigateToLoginScreen()
+                }
+            }
+        }
+    }
```

Comment:
r: flow는 생명주기 인식을 하지 못해서 메모리 누수가 발생할 수 있어요

```suggestion
lifecycleScope.launch {
    repeatOnLifecycle(Lifecycle.State.STARTED) {
        viewModel.uiEvent.collect { event ->
            when (event) {
                CommonEvent.TokenExpiration -> navigateToLoginScreen()
            }
        }
    }
}
```